tests/jsmarionette/runner/marionette-js-runner/host/session.js

```diff
@@ -31,10 +31,21 @@ Session.prototype = {
 
     return this.host.request('/get_crash_info', request)
     .then(function(info) {
+      // If info is empty, we didn't crash the process. We threw an error
+      // in JavaScript code instead.
+      if (Object.keys(info).length === 0) {
+        // The error we were passed originally will suffice in this case.
+        return err;
```

Comment:
Maybe you could also "return err" in a .catch() handler at the end of the chain ? So that if `parseCrashInfoXXX` fails we still have something ?

BW what is `err` useful in case of a real crash ? Is it falsy ? If it is you could maybe check this earlier ?